### PR TITLE
GitHub code synchronization toggle

### LINKED ISSUE #3426: [Caloris 2][Github Code] connection toggle for code sync

### PR DESCRIPTION
**Pull request by spolu**
## Description

Fixes #3426

- Removes bot_enabled front endpoint in favor of a generic config endpoint
- Moves slack bot toggle on this new endpoint
- Introduces Github code syncing toggle

Screenshot from 2024-02-05 00-07-07

## Risk

N/A

## Deploy Plan

- deploy `connectors` `front`

### CODE REVIEW

File: front/pages/api/w/[wId]/data_sources/[name]/managed/config/[key]/index.ts [hunk 1]
```diff
@@ -10,18 +10,18 @@ import { Authenticator, getSession } from "@app/lib/auth";
 import logger from "@app/logger/logger";
 import { apiError, withLogging } from "@app/logger/withlogging";
 
-export const PostBotEnabledRequestBodySchema = t.type({
-  botEnabled: t.boolean,
+export const PostManagedDataSourceConfigRequestBodySchema = t.type({
+  configValue: t.string,
```
Comment (PopDaph) [inline front/pages/api/w/[wId]/data_sources/[name]/managed/config/[key]/index.ts:14]: Why string over boolean?

File: front/pages/w/[wId]/builder/data-sources/[name]/index.tsx [hunk 2]
```diff
@@ -638,6 +642,82 @@ function SlackBotEnableView({
...
+        title="Code Synchronization"
+        visual={<ContextItem.Visual visual={GithubLogo} />}
+        action={
+          <div className="relative">
+            <SliderToggle
+              size="xs"
+              onClick={async () => {
+                await handleSetCodeSyncEnabled(!codeSyncEnabled);
+              }}
+              selected={codeSyncEnabled || false}
```
Comment (PopDaph) [inline front/pages/w/[wId]/builder/data-sources/[name]/index.tsx:705]: Why the default to false?

**Review by PopDaph — APPROVED** [review]

File: front/pages/api/w/[wId]/data_sources/[name]/managed/config/[key]/index.ts (hunk 1 above)
Comment (spolu) [reply front/pages/api/w/[wId]/data_sources/[name]/managed/config/[key]/index.ts:14]: A config is a string as per ConnectorsAPI :+1:

File: front/pages/w/[wId]/builder/data-sources/[name]/index.tsx (hunk 2 above)
Comment (spolu) [reply front/pages/w/[wId]/builder/data-sources/[name]/index.tsx:705]: cargo culted might not be needed, checking.

Comment (spolu) [reply front/pages/w/[wId]/builder/data-sources/[name]/index.tsx]: (removed)

**Pull request merged**